### ISSUE TITLE
Lookup word in pinyin for Chinese dictionary

### ISSUE THREAD

**Issue by karpalig** [goldendict/goldendict#1367]
I think this is a well-known task, but the dictionary can't to do it yet, the search for wishes in this branch also did not find such a desire.

Thank you for your great contribution!

**Comment by nikita-moor** [comment]
Do you mean that GoldenDict should automatically translate "zhongguo" to 中国? It can be easily made on the dictionary side. For example, if you have an article about China, it may have several keys: 中国, zhōngguó, zhon1guo2, zhonguo, ㄓㄨㄥ ㄍㄨㄛˊ. All modern formats (StarDict, MDict, Slob) support such a feature.

**Comment by karpalig** [comment]
> Do you mean that GoldenDict should automatically translate "zhongguo" to 中国?

I do not think that should automatically translate but show Chinese character which have same pinyin. Like this example I was write "mala" in search field and get this output :
![image](https://github.com/goldendict/goldendict/assets/7493374/d30a6873-f242-43d5-a39c-fb84abaa9c2d)

So, here I expect to see the same output
![image](https://github.com/goldendict/goldendict/assets/7493374/e9bcc8a0-e5a1-450f-97c4-9ce42cf086c2)
